### ISSUE TITLE
[$500] Assign task - Task report appears and disappears in LHN when log in as user B

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#28194]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Log in with any Gmail account (User A)
3. On LHN, click on the FAB button and Assign Task
4. Enter a title or description and Assign the task to yourself
5. On Share Somewhere, select a conversation with another user you have access to (User B)
6. Send a message in the Task Report
7. Log in as the user that the task was shared with (User B)
8. Click on the "# Replies" link in the task
9. Open any other conversation in LHN

## Expected Result:
You can see the task report in LHN

## Actual Result:
Task report disappears in LHN and appears again when open the Task

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome
- [ ] MacOS / Desktop

**Version Number:** 1.3.74-0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/3831f198-487e-4df0-ae89-55ab729c309b

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e489eb74081df51b</li>
        <li>Upwork Job ID: 1714382657873100800</li>
        <li>Last Price Increase: 2023-10-17</li>
    </ul>
</details>

**Comment by JmillsExpensify** [comment]
Sorry what's the bug here? If you leave the task report is shouldn't show up in the LHN in focus mode.

**Comment by roksanaz** [comment]
@lanitochka17 @JmillsExpensify Could you confirm what's the expected result here: should the task report appear in LHN or not? Because now it appears only for the user that created the task - is it expected behavior?

**Comment by JmillsExpensify** [comment]
> @lanitochka17 @JmillsExpensify Could you confirm what's the expected result here: should the task report appear in LHN or not? Because now it appears only for the user that created the task - is it expected behavior?

The task report should appear in the LHN for user B because there is an unread message

**Comment by rezkiy37** [comment]
So, after the investigation I found the root cause of this behaviour. I think it is a backend issue. 

Basically, the app have a few levels of checks to show/hide a `report` within the **LHN**. One of them verifies that the **report's** `notificationPreference` is not `hidden`.

https://github.com/Expensify/App/blob/4d75c0c192b21f83814918ebc46f9045b0be9def/src/components/LHNOptionsList/OptionRowLHN.js#L93-L96

However, **B user**'s tasks are hidden until **A user** posts a message to those tasks. Since, the **Pusher** takes responsibilities to deliver updates, when **B user** sends a message, it should take into account that `notificationPreference` must be updated for the task report for **A user**. 

In short, the backend should update `notificationPreference` and send the update property.

<details><summary>Example when "notificationPreference" equals "always"</summary>
<p>

<img width="1728" alt="Screenshot 2023-10-31 at 14 14 33" src="https://github.com/Expensify/App/assets/57314631/7eaaabc5-a0cc-490a-8cdd-524711da9be4">

</p>
</details> 

cc @JmillsExpensify, @lanitochka17

**Comment by rezkiy37** [comment]
Also, I have to inform that starting tomorrow I have a short vocation until next Monday (6.11.2023). Feel free to left any comments, I will address them. See you soon 😉

**Comment by rezkiy37** [comment]
Melvin, no "overdue" here.

**Comment by JmillsExpensify** [comment]
@thienlnam I wanted to get your thoughts on [this comment](https://github.com/Expensify/App/issues/28194#issuecomment-1787224853)? Do you agree with that? If so, I'll triage internally so someone internal can work on it.

**Comment by thienlnam** [comment]
This actually looks expected due to the new way we are handling threaded reports by 'hiding' them if you don't comment on it.

So, if a task is created unless User B actively participates in the task then it will not be shown in the LHN. If User B is assigned to the task, they should see the task in the LHN (that is getting updated in a different PR).

**Comment by rezkiy37** [comment]
@thienlnam, yes, it looks like that. Therefore no bug?

cc @JmillsExpensify

**Comment by thienlnam** [comment]
Yup correct - closing this out
